packages/components/src/ResourceList/Resource/Resource.component.js

```diff
@@ -56,33 +56,38 @@ function Resource({ parent, index, style, className, t }) {
 			aria-label={name}
 			onClick={onRowClick}
 		>
-			{icon && <Icon name={icon} />}
-			<div className={classNames('data-container', theme['data-container'])}>
-				<span className={classNames('title', theme.title)}>{name}</span>
-				{author && subtitle === undefined && (
-					<small className={classNames('author', theme.author)}>
-						{getAuthorLabel(t, author, modified)}
-					</small>
-				)}
+			{typeof as === 'function' ? (
+				as(rowData)
+			) : (
+				<React.Fragment>
+					{icon && <Icon name={icon} />}
+					<div className={classNames('data-container', theme['data-container'])}>
+						<span className={classNames('title', theme.title)}>{name}</span>
+						{author && subtitle === undefined && (
+							<small className={classNames('author', theme.author)}>
+								{getAuthorLabel(t, author, modified)}
+							</small>
+						)}
 
-				{subtitle !== undefined && !author && (
-					<small className={classNames('subtitle', theme.subtitle)} title={subtitle}>
-						{subtitle}
-					</small>
-				)}
-			</div>
-
-			<div className={classNames('flags-container', theme['flags-container'])}>
-				{Object.keys(FLAGS).map((flag, flagIndex) => (
-					<Icon
-						className={classNames(theme.flag, {
-							[theme.visible]: flags.includes(flag),
-						})}
-						key={flagIndex}
-						name={FLAGS[flag]}
-					/>
-				))}
-			</div>
+						{subtitle !== undefined && !author && (
+							<small className={classNames('subtitle', theme.subtitle)} title={subtitle}>
+								{subtitle}
+							</small>
+						)}
+					</div>
+					<div className={classNames('flags-container', theme['flags-container'])}>
```

Comment:
you could use getTheme to avoid declaration duplication of the class like there
https://github.com/Talend/ui/blob/c499be7ec6de1fb4bf00ba0de9cf882cefde1af1/packages/components/src/PieChart/PieChartIcon.component.js#L122